### PR TITLE
doxygen: Add libretro.h header

### PR DESCRIPTION
**Pull request by RobLoach**

### CODE REVIEW

File: libretro-common/include/libretro.h [hunk 1]
```diff
@@ -1,8 +1,13 @@
-/* Copyright (C) 2010-2020 The RetroArch team
+/**
+ * libretro.h is a simple API that allows for the creation of games and emulators.
+ *
+ * @file libretro.h
+ * @version 1
+ * @author libretro
+ * @copyright Copyright (C) 2010-2020 The RetroArch team
```
Comment (JesseTG) [inline libretro-common/include/libretro.h]: Don't forget to update the copyright to 2023.

File: libretro-common/include/libretro.h [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
  *
- * ---------------------------------------------------------------------------------------
- * The following license statement only applies to this libretro API header (libretro.h).
- * ---------------------------------------------------------------------------------------
+ * @paragraph LICENSE
+ * Copyright (C) 2010-2020 The RetroArch team
```
Comment (JesseTG) [inline libretro-common/include/libretro.h]: Don't forget to update the copyright to 2023.

**Review by JesseTG — CHANGES_REQUESTED** [review]

File: libretro-common/include/libretro.h (hunk 1 above)
Comment (RobLoach) [reply libretro-common/include/libretro.h]:
``` suggestion
 * @copyright Copyright (C) 2010-2023 The RetroArch team
 * ```

File: libretro-common/include/libretro.h (hunk 2 above)
Comment (RobLoach) [reply libretro-common/include/libretro.h]:
``` suggestion
 * Copyright (C) 2010-2023 The RetroArch team
```

**Pull request merged**